kernel/mmanutils.c

```diff
@@ -536,6 +579,52 @@ int myst_release_process_mappings(pid_t pid)
     return ret;
 }
 
+int proc_pid_maps_vcallback(myst_buf_t* vbuf)
+{
+    int ret = 0;
+    pid_t pid = myst_getpid();
+    struct locals
+    {
+        char maps_entry[PATH_MAX];
```

Comment:
file pathname string can be as long as PATH_MAX. map_entry won't be able to fit the max pathname, with the extra memory info in the front. Should maps_entry length be defined as PATH_MAX plus the worst case length of the extra memory info?

Reply:
Good point! Fixed.